tests/rest/client/utils.py

```diff
@@ -631,7 +658,13 @@ def initiate_sso_login(
 
         # hit the redirect url again with the right Host header, which should now issue
         # a cookie and redirect to the SSO provider.
-        location = channel.headers.getRawHeaders("Location")[0]
+        def get_location() -> str:
+            location_values = channel.headers.getRawHeaders("Location")
```

Comment:
Does this properly pull in `channel` again when it gets assigned on line 669? Might be clearer to just take it as an input argument.

Reply:
It does, but I had to confirm:

```python
>>> x = "a"
>>> def foo(): print(x)
... 
>>> foo()
a
>>> x = "b"
>>> foo()
b
```

That said, I prefer your clearer approach!